File: demo/java-demo/src/main/java/com/alibaba/testable/demo/DemoTemplate.java
package com.alibaba.testable.demo;

import java.util.*;

public class DemoTemplate {

    private <T> List<T> getList(T value) {
        List<T> l = new ArrayList<>();
        l.add(value);
        return l;
    }

    private <K, V> Map<K, V> getMap(K key, V value) {
        Map<K, V> m = new HashMap<>();
        m.put(key, value);
        return m;
    }

    public String singleTemplateMethod() {
        List<String> list = getList("demo");
        return list.get(0);
    }

    public String doubleTemplateMethod() {
        Map<String, String> map = getMap("hello", "testable");
        return map.get("hello");
    }

    public Set<?> newTemplateMethod() {
        Set<String> set = new HashSet<>();
        set.add("world");
        return set;
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/testable/demo/DemoTemplate.kt
package com.alibaba.testable.demo

import java.util.ArrayList
import java.util.HashMap
import java.util.HashSet

class DemoTemplate {
    private fun <T> getList(value: T): List<T> {
        val l: MutableList<T> = ArrayList()
        l.add(value)
        return l
    }

    private fun <K, V> getMap(key: K, value: V): Map<K, V> {
        val m: MutableMap<K, V> = HashMap()
        m[key] = value
        return m
    }

    fun singleTemplateMethod(): String {
        val list = getList("demo")
        return list[0]
    }

    fun doubleTemplateMethod(): String? {
        val map = getMap("hello", "testable")
        return map["hello"]
    }

    fun newTemplateMethod(): Set<*> {
        val set: MutableSet<String> = HashSet()
        set.add("world")
        return set
    }
}
